File: src/opentimelineio/timedText.cpp
#include "opentimelineio/timedText.h"

namespace opentimelineio {
    namespace OPENTIMELINEIO_VERSION {

        TimedText::TimedText(RationalTime const &in_time,
                             RationalTime const &out_time)
                : Parent() {
            set_marked_range(TimeRange::range_from_start_end_time(in_time, out_time));
        }

        TimedText::~TimedText() {}

        bool TimedText::read_from(Reader &reader) {
            return reader.read("texts", &_texts) &&
                   reader.read("style_ids", &_styleIDs) &&
                   Parent::read_from(reader);
        }

        void TimedText::write_to(Writer &writer) const {
            Parent::write_to(writer);
            writer.write("texts", _texts);
            writer.write("style_ids", _styleIDs);
        }

    }
}

File: src/py-opentimelineio/opentimelineio/schema/timed_text.py
from .. core._core_utils import add_method
from .. import _otio


@add_method(_otio.TimedText)
def __str__(self):
    return (
        'TimedText('
        '"{}", {}, {}, {})' .format(
            self.in_time,
            self.out_time,
            self.texts,
            self.style_ids
        )
    )


@add_method(_otio.TimedText)
def __repr__(self):
    return (
        'TimedText('
        'in_time={}, '
        'out_time={}, '
        'texts={}, '
        'style_ids={}'
        ')' .format(
            repr(self.in_time),
            repr(self.out_time),
            repr(self.texts),
            repr(self.style_ids)
        )
    )
